pytest-embedded/pytest_embedded/dut.py

```diff
@@ -131,6 +141,7 @@ def expect_exact(self, pattern, **kwargs) -> Match:  # noqa
             timeout (float): would raise `pexpect.TIMEOUT` exception when pattern is not matched after timeout
             expect_all (bool): need to match all specified patterns if this flag is `True`.
                 Otherwise match any of them could pass
+            not_matching (List[str]): patterns that should not appear in the buffer before expected patterns
```

Comment:
```suggestion
            not_matching: string, or compiled regex, or a list of string and compiled regex.
```

The `not_matching` keyword should accept all these three types of arguments.